File: src/main/java/MissingNumber.java
import java.util.Arrays;
import java.util.Scanner;

/**
 * https://cses.fi/problemset/task/1083
 */
public class MissingNumber {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        var n = input.nextInt();
        var allSum = 0;
        for (var i = 1; i <= n; i++) {
            allSum += i;
        }
        input.nextLine();
        var sum = Arrays.stream(input.nextLine().split(" "))
            .mapToInt(e -> Integer.parseInt(e))
            .sum();
        System.out.print(allSum - sum);
    }
}

File: src/bin/missing_number.rs
use std::io;

fn main() {
    let mut line = String::new();
    io::stdin().read_line(&mut line).unwrap();
    let n = line.trim().parse::<i32>().unwrap();
    let mut all_sum = 0;
    let mut i = 1;
    while i <= n {
        all_sum += i;
        i += 1;
    }
    let mut line = String::new();
    io::stdin().read_line(&mut line).unwrap();
    let sum: i32 = line
        .trim()
        .split(" ")
        .map(|e| e.parse::<i32>().unwrap())
        .sum();
    println!("{}", all_sum - sum);
}
